### PR TITLE
Migrating to go modules

### LINKED ISSUE #1392: Migrating to go modules
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind feature

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
/assign @zhujian7

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #1392 

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @iawia002 @hyy0322 

**Release note**:

```release-note
Migrating to go modules
```

### CODE REVIEW

File: pkg/server/biz/scm/gitlab/gitlab.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 	"github.com/caicloud/nirvana/log"
 	"github.com/xanzy/go-gitlab"
 	"golang.org/x/oauth2"
-	v4 "gopkg.in/xanzy/go-gitlab.v0"
+	v3 "gopkg.in/xanzy/go-gitlab.v0"
```
[the remark this answers is not in the document]
Comment (supereagle) [reply pkg/server/biz/scm/gitlab/gitlab.go]: `github.com/xanzy/go-gitlab` is imported 2 times. The originial is for `v4` and this one is for `v3`?

[the remark this answers is not in the document]
Comment (iawia002) [reply pkg/server/biz/scm/gitlab/gitlab.go]: > Why change the package alias to `v3`?

Go mod can't introduce `gopkg.in/xanzy/go-gitlab.v0` who's version higher than `v0.11.5`, so the strategy for now is using `gopkg.in/xanzy/go-gitlab.v0` as `v3`, `github.com/xanzy/go-gitlab` as `v4`. eg:

```
github.com/xanzy/go-gitlab v0.28.0
gopkg.in/xanzy/go-gitlab.v0 v0.6.0
```

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,59 @@
+module github.com/caicloud/cyclone
+
+go 1.12
```
Comment (iawia002) [inline go.mod]: Upgrade your local Golang, this should be `go 1.13`

File: Makefile [hunk 3]
```diff
@@ -121,6 +124,7 @@ build-linux:
 	  -e GOARCH=amd64                                                                  \
 	  -e GOPATH=/go                                                                    \
 	  -e CGO_ENABLED=0                                                                 \
+	  -e GOFLAGS=$(GOFLAGS)                                                            \
 	  -e SHELLOPTS=$(SHELLOPTS)                                                        \
 	  $(BASE_REGISTRY)/golang:1.12.12-stretch                                          \
```
[the remark this answers is not in the document]
Comment (zhujian7) [reply Makefile]: Bumping Golang version will submit by another PR.

File: go.mod (hunk 2 above)
Comment (zhujian7) [reply go.mod]: Same as above
@iawia002 PTAL

Comment (iawia002) [reply go.mod]: OK then

**Pull request merged**